chaoscenter/web/src/views/ChaosFault/__tests__/ChaosFault.test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import '@testing-library/jest-dom/';
+import type { FaultDetails } from '@api/core';
+import type { PredefinedExperiment } from '@api/entities';
+import ChaosFaultView from '../ChaosFault';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: () => ({ hubID: '123', faultName: 'faultTest' }),
+  useSearchParams: () => new URLSearchParams('hubName=testHub&isDefault=true')
+}));
+
+jest.mock('@hooks', () => ({
+  ...jest.requireActual('@hooks'), // if you want to keep other hooks unmocked
+  useAppStore: () => ({ projectID: 'mock-project-id' }),
+  useRouteWithBaseUrl: () => ({
+    toChaosHubs: () => '/chaoshubs',
+    toChaosHub: () => '/chaoshub',
+    toRoot: () => '/'
+  })
+}));
+
+jest.mock('@strings', () => ({
+  useStrings: () => ({ getString: (key: any) => key })
+}));
+
+describe('<ChaosFaultView />', () => {
```

Comment:
Since this test is specific to a route, after wrapping the render in test wrapper and default layout, use a before all router push to navigate to that specific route and then carry on the test

```tsx
beforeAll(() =>
  window.history.pushState(
    {},
    'Chaos Fault',
    '/chaos-hubs/6f39cea9-6264-4951-83a8-29976b614289/fault/aws/ecs-instance-stop'
  )
);
```